### DISCUSSION TITLE
Installeren op Pi 1 rev B

### DISCUSSION THREAD

**Discussion by devano1701** [dsmrreader/dsmr-reader#1789] [Questions & answers / How-to's] [answered]
### Description

Ik heb geen issue maar een vraag. 
Ik heb een oude Raspberry Pi (1) model B rev 2.
Het leek mij leuk om daar DSMR op te installeren.
Is dat überhaupt mogelijk? Mij is het nog niet gelukt.

Docker installeren lukt wel, maar docker-compose niet.
De stapsgewijze installatie lijkt wel te slagen; het uitlezen van de P1 poort werkt. Alle drie processen zijn running. 

Maar ik krijg een 504 fout van de webserver.
Dit suggereert dat nignx wel werkt als gateway, maar de achterliggende server niet goed functioneert.

Is het op een of andere manier mogelijk deze RPi hiervoor te gebruiken? 

### DSMR-reader version

laatste (geïnstalleerd via Installatie: Stapsgewijs)

### DSMR-reader platform

Native (e.g. manual installation)

**Comment by dennissiemensma** [comment, marked as answer]
De Pi 1 is vermoedelijk te traag voor dit project, gezien je een 504 error (timeout) krijgt. Ik denk dat een Pi 3 minimaal is voor een OK-ervaring en een Pi 4 of vergelijkbaar voor een goede ervaring. 

Echter heb ik geen harde minimale hardware specs voor je. Al kan ik me wel vaag herinneren dat bijvoorbeeld een Pi 2 het een paar jaar terug ook niet helemaal kon bijbenen.

Wat je wel kan doen is op een Pi 1/2 alleen een remote datalogger draaien. Dat proces is niet zo zwaar, maar dan zul je nog steeds ergens anders je DSMR-reader installatie moeten draaien, die dan de data van die remote datalogger onvangt.

**Reply by devano1701** [reply]
Dank voor je reactie. 
Ik begin ook te vermoeden dat de Pi1 niet snel genoeg was. Ik heb het ook met een Pi3 (even uit een ander projectje gehaald) geprobeerd en dat werkt wel (het installeren ging ook veel sneller).
Wel jammer, ik had die Pi1 over en hoopte op een truc. Deze alleen gebruiken als datalogger klinkt mooi, maar daarvoor heb ik te weinig kennis en kunde.

**Reply by devano1701** [reply]
RTFM: ik zie dat in de docs eea over remote loggen staat. Ik ga dat eens uit proberen.
Nogmaals bedankt voor de tip.

**Reply by dennissiemensma** [reply]
De remote datalogger is vooral een workaround voor als je DSMR-reader buiten je huis draait. Ik zou die niet gebruiken "om het gebruiken", want het script is hetzelfde als de ingebouwde datalogger.

**Reply by devano1701** [reply]
Ik begrijp je opmerking. Ik zal even mijn situatie toelichten.

Ik woon in een appartementencomplex en lees mijn eigen meter uit met DSMR in docker (xirixiz) op een Pi3. De data gaat over mijn netwerk via mosquitto naar HomeAssistant ook in docker op een Pi4. Samen met info uit een S0tool heb ik zo in HA een mooi overzicht van mijn energieverbruik. Dit werkt allemaal prima.  

Maar in het appartementencomplex hebben we ook een meter voor de gemeenschappelijke voorzieningen. Nu heb ik die Pi1 over en dacht ik daarmee die meter ook uit te kunnen lezen. Maar dat ging dus niet door de te trage Pi1. De Pi3 waarmee ik getest heb zat in een andere toepassing (met 7” screen) en moet eigenlijk weer terug. 

Dan nu mijn vraag:
Stel ik zet op de Pi1 alleen de datalogger. Dat moet dan mogelijk zijn, toch? 
 Waar zet ik dan de DSMR-reader? Mijn idee is op de Pi3 in een andere docker container. Ik verwijder de verwijzing naar device ttyUSB0 zodat ik geen conflict krijg met mijn eigen meter. Docker containers zijn toch gescheiden van elkaar? Ik zie evenwel dat de database in een aparte container draait. Hoe zorg ik er nu voor dat de nieuwe DSMR-reader een eigen database krijgt. 

Hopelijk kan je mij een beetje op weg helpen. Ik zou het in ieder geval op prijs stellen.

**Reply by dennissiemensma** [reply]
Alleen een remote datalogger op de Pi 1 moet prima werken. Het is in principe 1 proces (behalve het OS zelf) wat constant de meter uitleest en, na elk volledig telegram, het complete telegram naar de API van "een" DSMR-reader installatie stuurt. En dat voor altijd blijft doen. Je zult dus wel een netwerkverbinding moeten hebben van de Pi 1 naar de Pi 3.

Voor het maken van een tweede DSMR-reader installatie kun je die als het goed is naast elkaar draaien. De ontvangende installatie draai je dan in `api_server` modus:
- https://github.com/xirixiz/dsmr-reader-docker#dsmr-datalogger-related

Je kunt zelfs nog proberen om op die Pi 1 Docker te draaien en dan ook daar DSMR-reader, in `api_client` modus. Alleen weet ik niet of dat performt. Scheelt je wel hoop installatiestappen.

----

Ik zou zelf voor de tweede DSMR-reader installatie gewoon een tweede database-instantie ernaast draaien. Dan heb je ze ook mooi gescheiden. Je stelt dan op de ontvangende DSMR-reader de andere database credentials in. Ik adviseer wel voor beide installaties een andere DB-gebruiker en DB-naam te gebruiken, zodat je ze nooit per ongeluk door elkaar kan halen.

Als voorbeeld de dev-stack die ik gebruik voor ontwikkelen (niet bedoeld voor productie), die heeft bijvoorbeeld ook 2 x Postgres en 1 x MySQL. Zou moeten werken naast elkaar:
- https://github.com/dsmrreader/dsmr-reader/blob/v5/dsmrreader/provisioning/docker/docker-compose.dev.yml#L33-L37
- https://github.com/dsmrreader/dsmr-reader/blob/v5/dsmrreader/provisioning/docker/docker-compose.dev.yml#L74


En 
- https://github.com/dsmrreader/dsmr-reader/blob/v5/dsmrreader/provisioning/docker/docker-compose.dev.yml#L40-L44
- https://github.com/dsmrreader/dsmr-reader/blob/v5/dsmrreader/provisioning/docker/docker-compose.dev.yml#L94
